pkg/networkserver/grpc_deviceregistry.go

```diff
@@ -565,6 +565,9 @@ func (ns *NetworkServer) Delete(ctx context.Context, req *ttnpb.EndDeviceIdentif
 	}
 	var evt events.Event
 	_, _, err := ns.devices.SetByID(ctx, req.ApplicationIdentifiers, req.DeviceID, nil, func(ctx context.Context, dev *ttnpb.EndDevice) (*ttnpb.EndDevice, []string, error) {
+		if dev == nil || !dev.ApplicationIdentifiers.Equal(req.ApplicationIdentifiers) {
```

Comment:
```suggestion
		if dev == nil {
```
https://github.com/TheThingsNetwork/lorawan-stack/pull/2343#discussion_r408112582

Reply:
Tried to mirror the behaviour of the Join Server:

https://github.com/TheThingsNetwork/lorawan-stack/blob/0da92e3acedf7e8003b2550c4f3b4d5b7872dacb/pkg/joinserver/grpc_deviceregistry.go#L289

Removed if it does not make sense in the context of the Network Server.

Reply:
Yeah, that check makes no sense either anymore it should have been removed in https://github.com/TheThingsNetwork/lorawan-stack/commit/7f5ab3e4fa52db573902af96f147706a7f097e61 when we switched from keying by EUIs to keying by IDs